lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1949,6 +1949,7 @@ bool TypeCheckASTNodeAtLocRequest::evaluate(Evaluator &evaluator,
   if (auto CE = dyn_cast<ClosureExpr>(DC)) {
     if (CE->getBodyState() == ClosureExpr::BodyState::Parsed) {
       swift::typeCheckASTNodeAtLoc(CE->getParent(), CE->getLoc());
+      checkClosureActorIsolation(CE);
```

Comment:
I'm concerned that `ActorIsolationChecker` walk into unrelated sibling closures or the children of this closure.
How about exposing `ActorIsolationChecker::determineClosureIsolation()` and call it directly?
```cpp
ActorIsolationChecker checker(closure->getParent());
CE->setActorIsolation(checker->determineClosureIsolation(CE));
```

Reply:
Or, how about calling `checkFunctionActorIsolation()` (or `checkTopLevelActorIsolation()`) in `ide::typeCheckContextAt()` right after calling `typeCheckASTNodeAtLoc()`?

Reply:
Thanks. I didn’t realize that `TypeCheckASTNodeAtLocRequest` was called multiple times. 

I think it’ll be good to just set the current closure’s actor isolation as you proposed in the first comment because it minimizes the amount of actor isolation checking that needs to be done. I’ll update it.